plugin/pkg/policy/engine.go

```diff
@@ -0,0 +1,71 @@
+package policy
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/request"
+)
+
+const (
+	// TypeNone a no value
+	TypeNone = byte(iota)
+	// TypeRefuse policy action is REFUSE (do not resolve a query and return rcode REFUSED)
+	TypeRefuse
+	// TypeAllow policy action is ALLOW (continue to resolve query)
+	TypeAllow
+	// TypeBlock policy action is BLOCK (do not resolve a query and return rcode NXDOMAIN)
+	TypeBlock
+	// TypeDrop policy action is DROP (do not resolve a query and simulate a lost query)
+	TypeDrop
+
+	// TypeCount total number of actions allowed
+	TypeCount
+)
+
+const (
+	nameTypeNone   = "none"
+	nameTypeAllow  = "allow"
+	nameTypeRefuse = "refuse"
+	nameTypeBlock  = "block"
+	nameTypeDrop   = "drop"
+)
+
+// NameTypes keep a mapping of the byte constant to the corresponding name
+var NameTypes [TypeCount]string
+
+func initNameTypes() {
+	NameTypes[TypeNone] = nameTypeNone
```

Comment:
NameType is only use in the setup function of firewall, why is all this stuff needed and exported?

Reply:
Engine is an API for policy engines. The TypesXXX are predefined. 
I think that the names of these types should be visible for these policy engines that will implement Engine.